src/pymovements/gaze/gaze_dataframe.py

```diff
@@ -239,23 +245,28 @@ def __init__(
         self.frame = data
 
         self.trial_columns = [trial_columns] if isinstance(trial_columns, str) else trial_columns
+        self.experiment = experiment
 
         # In case the 'time' column is already present we don't need to do anything.
-        # Otherwise, create a new time column starting with zero.
+        # Otherwise, create a new time column starting with zero and set time unit to steps
         if time_column is None and 'time' not in self.frame.columns:
-            # In case we have an experiment with sampling rate given, we convert to milliseconds.
+            # In case we have an experiment with sampling rate given, we create a time
             if experiment is not None and experiment.sampling_rate is not None:
-                sampling_rate_factor = 1000 / experiment.sampling_rate
-            else:
-                sampling_rate_factor = 1
+                self.frame = self.frame.with_columns(
+                    time=pl.arange(0, len(self.frame)),
+                )
 
-            self.frame = self.frame.with_columns(
-                time=pl.arange(0, len(self.frame)) * sampling_rate_factor,
-            )
+                time_column = 'time'
+                time_unit = 'step'
+            else:
+                warnings.warn(
```

Comment:
if no time column is specified and it is not possible to generate timestamps based on sampling rate there is a warning. Now all tests not using a time column in GazeDataFrame are failing. Can you give me an advise on how to handle this?

Reply:
oh what a bummer.. do you know if it affects more than `gaze_transform_test.py`?

I think the only real solution is adding an experiment definition to test cases where we don't want any warnings.

In the `gaze_transform_test.py` this definition here is used a lot due to simple trigonometrics:

```python
 pm.Experiment(
    screen_width_px=100,
    screen_height_px=100,
    screen_width_cm=100,
    screen_height_cm=100,
    distance_cm=100,
    origin='center',
    sampling_rate=1000,
)
```

Maybe you could create a variable at the top of the file named `experiment_100` (there's an `experiment` fixture already, so don't use that name) and reuse the definition in all the test cases in that file.

Sorry for that annoying work.

Reply:
you will probably need `experiment_100_center?` and `experiment_100_lower_left` though.

Reply:
Thanks for your response, i'll look into it!

Reply:
Most tests `gaze_init_test.py` using a gaze dataframe without time column fail too. I don't really think just adding a time column or experiment in every test where it is missing is a good solution as the time has no effect on the data that is being tested. So i think it is just blowing up the test cases. 

And for example here we have a test that is a legit case but would trigger a warning:
`        pytest.param(
            {
                'data': pl.from_dict(
                    {'x': [1.23, 2.34, 3.45], 'y': [4.56, 5.67, 6.78]},
                    schema={'x': pl.Float64, 'y': pl.Float64},
                ),
                'position_columns': ['x', 'y'],
            },
            pl.from_dict(
                {'position': [[1.23, 4.56], [2.34, 5.67], [3.45, 6.78]]},
                schema={'position': pl.List(pl.Float64)},
            ),
            2,
            id='df_three_rows_two_position_columns_no_time_no_experiment',
`


Couldn't we do something like:

`
def test_init_gaze_dataframe_has_expected_attrs(init_kwargs, expected_frame, expected_n_components):
    with warnings.catch_warnings():
        warnings.simplefilter('ignore', UserWarning)

        gaze = pm.GazeDataFrame(**init_kwargs)
        assert_frame_equal(gaze.frame, expected_frame)
        assert gaze.n_components == expected_n_components
`

Reply:
I see, filtering the warnings is probably the best approach then.

Instead of using warnings.simplefilter, please use [warnings.filterwarnings](https://docs.python.org/3/library/warnings.html#warnings.filterwarnings) and set the expected message and module to not ignore more warnings than necessary.